internal/feed/reddit.go

```diff
@@ -30,8 +30,13 @@ type subredditResponseJson struct {
 	} `json:"data"`
 }
 
-func FetchSubredditPosts(subreddit string, commentsUrlTemplate string) (ForumPosts, error) {
-	requestUrl := fmt.Sprintf("https://www.reddit.com/r/%s/hot.json", url.QueryEscape(subreddit))
+func FetchSubredditPosts(subreddit string, commentsUrlTemplate string, customEndpoint string) (ForumPosts, error) {
+	subreddit = url.QueryEscape(subreddit)
+	requestUrl := fmt.Sprintf("https://www.reddit.com/r/%s/hot.json", subreddit)
+	if customEndpoint != "" {
+		strings.ReplaceAll(customEndpoint, "{SUBREDDIT}", subreddit)
+		requestUrl = customEndpoint
```

Comment:
```suggestion
		requestUrl = strings.ReplaceAll(customEndpoint, "{REQUEST-URL}", requestUrl)
```

The reason I initially suggested replacing the entire URL is because I'm going to add the option to sort by new and potentially also the option to search for specific keywords:

```yaml
- type: reddit
  subreddit: selfhosted
  sort-by: new
  search: dashboard
```

If the user was to proxy just the subreddit like this:

```yaml
- type: reddit
  subreddit: selfhosted
  custom-endpoint: https://proxy/r/{SUBREDDIT}/hot.json
  sort-by: new       
  search: dashboard  
```

Then the `sort-by` and `search` would no longer work and it's up to the user to figure out how to implement them via the URL. If they were to provide the custom-endpoint like this then everything will work as expected:

```yaml
- type: reddit
  subreddit: selfhosted
  custom-endpoint: https://proxy/{REQUEST-URL}
  sort-by: new       
  search: dashboard 
```

This is also how cors-anywhere works by default, it takes the full URL as the path.

Reply:
> This is also how cors-anywhere works by default, it takes the full URL as the path.

Ah, I was using another service myself that I wish to keep private so it doesn't get patched.

Actually it's quite similar, I do agree this may better. I'll amend the PR later.